parsers/US_PJM.py

```diff
@@ -164,40 +165,34 @@ def fetch_production(
         "datetime_beginning_utc": target_datetime.strftime("%Y-%m-%dT%H:00:00.0000000"),
     }
     resp_data = _fetch_api_data(kind="gen_by_fuel", params=params, session=session)
-    print(resp_data)
 
-    data = pd.DataFrame(resp_data["items"])
-    if data.empty:
+    items = resp_data.get("items", [])
+
+    if items == []:
         raise ParserException(
             parser=PARSER,
             message=f"{target_datetime}: Production data is not available in the API",
             zone_key=zone_key,
         )
 
-    data["datetime_beginning_utc"] = pd.to_datetime(data["datetime_beginning_utc"])
-    data = data.set_index("datetime_beginning_utc")
-    data["fuel_type"] = data["fuel_type"].map(FUEL_MAPPING)
-
     production_breakdown_list = ProductionBreakdownList(logger)
-    for dt in data.index.unique():
-        production_mix = ProductionMix()
-        storage_mix = StorageMix()
-
-        data_dt = data.loc[data.index == dt]
-        for i in range(len(data_dt)):
-            row = data_dt.iloc[i]
-            if row["fuel_type"] == "battery":
-                storage_mix.add_value("battery", row.get("mw"))
+    for key, group in groupby(items, itemgetter("datetime_beginning_utc")):
+        dt = datetime.fromisoformat(key).replace(tzinfo=timezone.utc)
+        production = ProductionMix()
+        storage = StorageMix()
+        for data in group:
+            mode = FUEL_MAPPING[data["fuel_type"]]
+            value = data["mw"]
+            if mode == "battery":
+                storage.add_value(mode, value)
```

Comment:
It's possible that the value should be inverted here as that's what's done in the ENTSO-E parser. Need to double check this before we merge.